sonar-scanner-engine/src/main/java/org/sonar/batch/issue/IssueTransformer.java

```diff
@@ -93,9 +94,9 @@ public static TrackedIssue toTrackedIssue(BatchComponent component, ScannerRepor
     issue.setKey(Uuids.createFast());
     issue.setComponentKey(component.key());
     issue.setRuleKey(ruleKey);
-    issue.setGap(rawIssue.hasGap() ? rawIssue.getGap() : null);
+    issue.setGap(rawIssue.getGap() != 0 ? rawIssue.getGap() : null);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1244)